flagging_site/templates/index.html

```diff
@@ -1,30 +1,82 @@
+<p>
+    CRWA's Flagging Program uses a system of color-coded flags to indicate whether or not the river's water quality is
+    safe for boating at eleven boating locations between Watertown and Boston. Flag colors are based on E. coli and
+    cyanobacteria (blue-green algae) levels; blue flags indicate suitable boating conditions and red flags indicate
+    potential health risks.
+</p>
+<p>
+    CRWA estimates E. coli bacteria levels using models that take into account a variety of environmental factors
+    including rainfall and river flow. CRWA collects water samples weekly to verify the model forecasts. Cyanobacteria
+    monitoring is conducted every 1-2 weeks.
+</p>
+<br />
+<!-- Is this needed? CRWA's website links to a 2019 report using a button 
+<p>
+    Subscribe to water quality alerts through the sign up form below.
+</p> -->
+<!-- optional 2019 report button here? -->
+
+<h2>Water Quality Today</h2>
+<!-- Older language
+<p>
+    Water quality flags, with forecasted E.coli concentrations and
+    probabilities (updated hourly).
+</p> -->
+<!-- Newer language -->
+<p>
+    CRWA's water quality models are predicting blue flags for all boathouses.
+
+</p>
+<p>
+    In accordance with the MA Department of Public Health (DPH) cyanobacteria (blue-green algae) advisory posted for the
+    Charles River between the Museum of Science and the Mass Ave Bridge, red flags are being flown at Charles River
+    Yacht Club, Union Boat Club, Community Boating, and Paddle Boston at Kendall Square. Public health officials
+    recommend that people and pets avoid contact in areas of algae concentration and rinse thoroughly in the event of
+    contact. DPH will conduct follow-up sampling when the bloom is no longer evident. DPH requires at least two samples,
+    one week apart, to recommend rescinding the advisory.
+
+</p>
+<br /><br />
+<table class="home-reach">
+    {% for reach_no, reach in homepage.items() %}
+    <tr>
+        <td class="home-reach">
+            <div> Reach {{ reach_no }} </div>
+            <ul class="home-boat-house">
+                {% for attr, vals in reach.names.items() %}
+                <li> {{ vals }} </li>
+                {% endfor %}
+            </ul>
+        </td>
+        {% if reach.flag %}
+        <td class="home-reach"><img class="flag" src="{{ url_for('static', filename='images/blue_flag.jpg') }}" /></td>
         {% else %}
-            <img class="flag" src="{{ url_for('static', filename='images/red_flag.jpg') }}" />
+        <td><img class="flag home_reach" src="{{ url_for('static', filename='images/red_flag.jpg') }}" /></td>
         {% endif %}
-    </div>
+    </tr>
     {% endfor %}
-
+</table>
+{% for key, value in homepage.items() %}
+<p class="home-centered">
+    <!-- Currently uses time of last row for reach 5 table. Needs unit test to ensure this matches the times for reach 2, 3, and 4 -->
+    {% if loop.last %}
+    (Last updated: {{value.time}})
+    {% endif %}
```

Comment:
Wow! This is a very astute observation.

You are basically pointing to some of the limitations of the current code base, and you're doing a good job dealing with those limitations.

Your note about unit-testing this is totally correct and not something I had considered yet as part of unit-testing for the data. Good job! I'll make sure the unit-test writers for the data models include this as part of their tests.

For now, I know that the models all have the same final timestamp. It's something we need to test though, or guarantee in some other way.

Also for now, my suggestion is that we'll pass only _one_ time value in, since we're only using one anyway to begin with. Let's do this for now:

`(Last updated: {{model_last_updated_time}}`